.buildkite/pipeline.yml

```diff
@@ -32,28 +29,33 @@ steps:
   #################
   - label: "🧹 Lint"
     command: |
-      .buildkite/install-dependencies.sh
-
+      bundle install
       echo "--- :rubocop: Run Rubocop"
       bundle exec rubocop
     env: *common_env
-    plugins: *common_plugins
+    plugins:
+      - docker#v3.8.0:
+          image: "ruby:2.7.4"
```

Comment:
I'm guessing you didn't create a YAML anchor for this one because the only other place you currently use it is on line 51 but you need to pass the API token on that other one so they end up differing?

I'd still suggest to:

 - Make line 71 (for the `gem-push` step) use the same `ruby:2.7.4` docker image than the one we use for linting and testing
 - Then get rid of the `*common_plugins` and `bash_cache` anchors as they would not be used anymore (we won't rely on the bash-cache docker image for release-toolkit, only on `ruby:2.7.4`)
 - Make new anchors for the ruby image:

```yaml
common_params:
  plugins:
  - &docker_plugin
    docker#v3.8.0:
      image: &ruby_version "ruby:2.7.4"
  - &docker_plugin_with_danger_token
    docker#v3.8.0:
      image: *ruby_version
      propagate-environment: true
      environment:
        - "DANGER_GITHUB_API_TOKEN"
```
That way, the docker plugin version as well as the ruby version we use are all centralized on top of the file, and we have less risk of forgetting to update everything at once next time we update those.

Then of course for the lint step you'd use `plugins: [*docker_plugin]` while for the danger step you'd use 
`plugins: [*docker_plugin_with_danger_token]`


PS: There are more advanced solutions in YAML to avoid repeating the `docker#3.8.0` key here (i.e ways to create an anchor just on the `docker#v3.8.0` key name, not the whole dict, etc), but I don't think it's worth getting into those. As long as all the versions we might want to update are grouped at the top in a centralized place this is good enough.

Reply:
Addressed in 538430e84ebdf3f4daa70ae8639bc665d9ba5f19 (followed by some noodling around to move the `gem push` stuff over to docker as well).